integration_tests/src/examples.rs

```diff
@@ -299,3 +176,113 @@ fn get_formated_body(body_str: &str) -> String {
         msg_string,
     )
 }
+
+pub enum TestSample {
+    /// 10 minute animated video with sound
+    BigBuckBunny,
+    /// 11 minute animated video with sound
+    ElephantsDream,
+    /// 28 sec video with no sound
+    Sample,
+    /// looped 28 sec video with no sound
+    SampleLoop,
+    /// generated sample video with no sound (also with second timer when using ffmpeg)
+    Generic,
+}
+
+#[derive(Debug)]
+struct AssetData {
+    url: String,
+    path: PathBuf,
+}
+
+fn download_all_assets() -> Result<()> {
+    let assets = [AssetData {
+        url: String::from("https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"),
+        path: examples_root_dir().join("examples/assets/BigBuckBunny.mp4"),
+    },
+    AssetData {
+        url: String::from("http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4"),
+        path: examples_root_dir().join("examples/assets/ElephantsDream.mp4"),
+    },
+    AssetData {
+        url: String::from("https://filesamples.com/samples/video/mp4/sample_1280x720.mp4"),
+        path: examples_root_dir().join("examples/assets/sample_1280_720.mp4"),
+    }];
+
+    let mut any_asset_downloaded = false;
+    for asset in assets {
+        match download_asset(&asset) {
+            Ok(()) => any_asset_downloaded = true,
+            Err(_) => info!("Error while downloading asset: {:?}", asset),
+        };
+    }
```

Comment:
```suggestion
    let any_asset_downloaded = assets.iter()
        .map(download_asset)
        .any(|download_result| download_result.is_ok());
```

Reply:
BTW shouldn't you check if all assets are downloaded correctly? If you have 100 assets, downloading 99 fails and only 1 pass, this won't throw an error

Reply:
nope, @wkozyra95 suggested that failure when downloading some assets shouldn't throw en error, it should only be thrown if user tries to use asset that failed. At least that's how I understood it, if I'm wrong correct me.

Reply:
Ok, makes sense